src/components/structures/InteractiveAuth.js

```diff
@@ -117,19 +157,35 @@ export default React.createClass({
 
     _renderCurrentStage: function() {
         const stage = this.state.authStage;
-        var StageComponent = getEntryComponentForLoginType(stage);
+        if (!stage) return null;
+
+        const StageComponent = getEntryComponentForLoginType(stage);
         return (
             <StageComponent ref="stageComponent"
                 loginType={stage}
+                matrixClient={this.props.matrixClient}
                 authSessionId={this._authLogic.getSessionId()}
+                clientSecret={this._authLogic.getClientSecret()}
                 stageParams={this._authLogic.getStageParams(stage)}
                 submitAuthDict={this._submitAuthDict}
                 errorText={this.state.stageErrorText}
                 busy={this.state.busy}
+                inputs={this.props.inputs}
+                stageState={this.state.stageState}
+                fail={this._onAuthStageFailed}
+                setEmailSid={this._setEmailSid}
+                makeRegistrationUrl={this.props.makeRegistrationUrl}
```

Comment:
Oh, potentially. It's true that it special cases electron to avoid giving out file: URIs, although really the link itself is platform independent since the link needs to work wherever you click on it. Hmm.